### ISSUE TITLE
Does validation set has an impact on the update of gradients?

### ISSUE THREAD

**Issue by gitmcdull** [pytorch/fairseq#146]
As the size of our data is not big, I want to use all my training data in training without validation. Because the train.py won't work if I did not specify --validpref, I use a subset of my training data as the validation set with --validpref. 

According to my knowledge, validation sets have no impact on the update of gradients, while sometimes they may cause early stop. But I found the learning process are quite different with different validation sets:

(1) Use a subset of training set as validation set
I found the training process took much more epochs.And the loss and validation loss get very small, e.g. epoch = 100, loss = 0.08, valid loss = 0.02

(2) Use other data as validation set
The training process will stop quite early, e.g. epoch = 10, loss = 0.6, valid loss = 0.07

I have two questions: 
A. Is it caused by early stop? 
B. Are the models from the training (1) reliable?
